Fix introduction and added transform indications (#57)
greenelab/knowledge-graph-review: content/04.network_applications.md

Following construction, $A$ is decomposed into the following three parts: a square matrix $Σ$ and a set of two small rectangular matrices $U$ and $V^{T}$.
Values within $Σ$ are called singular values, which are akin to eigenvalues [@doi:10.1007/BF02288367].
Each row in $U$ and each column in $V^{T}$ represents nodes within a low dimensional space [@doi:10.1007/BF02288367; @doi:10/bm8dnf].

> This low dimensional representation loses the concept of a knowledge graph as only nodes are explicitly represented in this context. 

Reviewer: Do you want to say that it loses the _concept_ or that it no longer meets the definition? In this case, is there some position thing you can say about the representation? What do you get for giving up the knowledge graph?

The adjacency and degree matrices are converted into a laplacian matrix ($L$), which is a matrix that shares the same properties as the adjacency matrix.
The laplacian matrix is generated by subtracting the adjacency matrix from the degree matrix ($L=D-A$) and, once constructed, the algorithm uses linear algebra to calculate the laplacian's eigenvalues and eigenvectors ($Lx = \lambda Dx$).
The generated eigenvectors represent the knowledge graph's nodes represented in a low dimensional space [@doi:10.1162/089976603321780317].
Other efforts have used variants of this algorithm to construct low dimensional representations of knowledge graphs [@raw:gongreplearn; @doi:10.1109/TKDE.2016.2606098; @arxiv:1905.09763].
Typically, eigenmaps work well when knowledge graphs have a sparse number of edges between nodes but struggle when presented with denser networks [@doi:10.1093/bioinformatics/btz718; @doi:10.1371/journal.pcbi.1005621; @arxiv:1905.09763].
An open area of exploration is to adapt these methods to accommodate knowledge graphs that have a large number of edges.

Matrix factorization is a powerful technique that represents high dimensional data in a low dimensional space.

> Knowledge graphs represented in this space no longer meets our definition of a knowledge graph; however, this representation permits heavy computation such as calculating similarity metrics (e.g., cosine similarity [@arxiv:1910.09129]) or running machine learning classifiers.

Reviewer: ```suggestion
The representation of a knowledge graph in this reduced space does not meet our definition of a knowledge graph; however, this representation supports many use cases including similarity-based (e.g., cosine similarity [@arxiv:1910.09129]) and machine learning applications.
```

The adjacency and degree matrices are converted into a laplacian matrix ($L$), which is a matrix that shares the same properties as the adjacency matrix.
The laplacian matrix is generated by subtracting the adjacency matrix from the degree matrix ($L=D-A$) and, once constructed, the algorithm uses linear algebra to calculate the laplacian's eigenvalues and eigenvectors ($Lx = \lambda Dx$).
The generated eigenvectors represent the knowledge graph's nodes represented in a low dimensional space [@doi:10.1162/089976603321780317].

> Similar to SVD, this representation loses the concept of a knowledge graph as only nodes are explicitly represented.

Reviewer: Same as above

It is possible to layer on additional objectives by modifying the loss function to take into account criteria above and beyond reconstruction loss [@arxiv:1312.6114; @arxiv:1802.03480].
In the context of knowledge graphs, the generated space correlates nodes with dense vectors that capture a graph's connectivity structure [@doi:10.1109/DSAA.2018.00034; @arxiv:1611.07308; @arxiv:1802.04407].
Despite the high potential of autoencoders, this method relies on an adjacency matrix for input which can run into scalability issues as a knowledge graph asymptotically increases in size [@doi:10.1109/TKDE.2019.2951398].
Plus, Khosla et al. discovered that approaches akin to node2vec outperformed algorithms using autoencoders when undergoing link prediction and node classification [@doi:10.1109/TKDE.2019.2951398].

Overall, the performance of neural network models largely depends upon the structure of nodes and edges within a knowledge graph [@doi:10.1109/TKDE.2019.2951398].
Furthermore, when these approaches are used only nodes are explicitly represented by these vectors.

> This means a represented knowledge graph no longer meets our definition of a knowledge graph; however, this new representation ameliorates the ability to perform heavy computation such as running a deep learning model for biomedical applications.

Reviewer: ```suggestion
This means a represented knowledge graph no longer meets our definition of a knowledge graph; however, this representation can make it more suitable for many biomedical applications.
```
